src/test/java/uk/ac/ebi/ampt2d/accession/service/MonotonicAccessionGeneratorTest.java

```diff
@@ -0,0 +1,313 @@
+        ContinuousIdBlock block = repository.findFirstByCategoryIdAndInstanceIdOrderByEndDesc(CATEGORY_ID, INSTANCE_ID);
+        assertEquals(-1, block.getLastCommitted());
+
+        long[] accessions2 = generator.generateAccessions(BLOCK_SIZE);
+        assertEquals(0, accessions2[0]);
+        assertEquals(1, accessions2[1]);
+        assertEquals(3, accessions2[2]);
+        assertEquals(4, accessions2[3]);
+        assertEquals(5, accessions2[4]);
+        assertEquals(8, accessions2[5]);
+        assertEquals(9, accessions2[6]);
+        assertEquals(10, accessions2[7]);
+        assertEquals(1000, accessions2[8]);
+        assertEquals(1092, accessions2[100]);
+
+        // Only 998 elements have been confirmed due to 999 not being confirmed, reread the block to assert it
+        generator.commit(accessions2);
+        block = repository.findOne(block.getId());
+        assertEquals(998, block.getLastCommitted());
+        // 999 is committed and then the remaining elements get confirmed
+        generator.commit(999);
+        block = repository.findFirstByCategoryIdAndInstanceIdOrderByEndDesc(CATEGORY_ID, INSTANCE_ID);
+        assertEquals(1991, block.getLastCommitted());
+    }
+
+    @Test
+    public void assertRecoverNoPendingCommit() throws Exception {
+        MonotonicAccessionGenerator generator = getMonotonicAccessionGenerator();
+        long[] accessions1 = generator.generateAccessions(BLOCK_SIZE);
+        // Now assume that the db layer has stored some elements and that the application has died and restarted.
+
+        MonotonicAccessionGenerator generatorRecovering = new MonotonicAccessionGenerator(BLOCK_SIZE, CATEGORY_ID,
+                INSTANCE_ID, repository);
+        generatorRecovering.afterPropertiesSet();
+
+        generatorRecovering.recoverState(new long[]{2, 3, 5});
+        ContinuousIdBlock block = repository.findFirstByCategoryIdAndInstanceIdOrderByEndDesc(CATEGORY_ID,
+                INSTANCE_ID);
+        assertEquals(-1, block.getLastCommitted());
+    }
+
+    @Test
+    public void assertRecoverPendingCommit() throws Exception {
+        MonotonicAccessionGenerator generator = getMonotonicAccessionGenerator();
+        long[] accessions1 = generator.generateAccessions(BLOCK_SIZE);
+        generator.commit(0,1);
+        // Now assume that the db layer has stored some elements and that the application has died and restarted.
+
+        MonotonicAccessionGenerator generatorRecovering = new MonotonicAccessionGenerator(BLOCK_SIZE, CATEGORY_ID,
+                INSTANCE_ID, repository);
+        generatorRecovering.afterPropertiesSet();
+
+        generatorRecovering.recoverState(new long[]{2, 3, 5});
+        ContinuousIdBlock block = repository.findFirstByCategoryIdAndInstanceIdOrderByEndDesc(CATEGORY_ID,
+                INSTANCE_ID);
+        assertEquals(3, block.getLastCommitted());
```

Comment:
I guess there won't be a case of non monotonic recovery ranges in same block , if so should the block.getLastCommited return 5 ?